index.bs

```diff
@@ -425,6 +434,56 @@ The <dfn attribute for=XRJointPose>radius</dfn> attribute returns the [=skeleton
 
 The user-agent MUST set {{XRJointPose/radius}} to an emulated value if the [=/XR device=] does not have the capability of determining this value, either in general or in the current [=XRSession/animation frame=] (e.g. when the [=skeleton joint=] is partially obscured).
 
+WebXR Device API Integration {#webxrintegration}
+============================
+
+{{XRSpace}} changes {#xrspacechanges}
+-------------------
+<div class="algorithm" data-algorithm="populate-the-pose-for-hands">
+
+This module replaces the steps given by "[=populate the pose=]" from the WebXR specification. Instead when the user agent will
+<dfn dfn-for="hands">populate the pose</dfn> of an {{XRSpace}} |space| in an {{XRSpace}} |baseSpace| at the time represented by an {{XRFrame}} |frame| into an {{XRPose}} |pose|, with an optional |force emulation| flag, the user agent MUST run the following steps:
+
+  1. If |frame|'s [=XRFrame/active=] boolean is `false`, throw an {{InvalidStateError}} and abort these steps.
+  1. Let |session| be |frame|'s {{XRFrame/session}} object.
+  1. If |space|'s [=XRSpace/session=] does not equal |session|, throw an {{InvalidStateError}} and abort these steps.
+  1. If |baseSpace|'s [=XRSpace/session=] does not equal |session|, throw an {{InvalidStateError}} and abort these steps.
+  1. Check if [=poses may be reported=] and, if not, throw a {{SecurityError}} and abort these steps.
+  1. Let |limit| be the result of whether [=poses must be limited=] between |space| and |baseSpace|.
+  1. Let |transform| be |pose|'s {{XRPose/transform}}.
+  1. If |space| is a {{XRJointSpace}}, perform the following steps:
+    1. Let |jointSpaces| be the result of calling "values()" on |space|'s [=XRJointSpace/hand=].
+    1. For each |jointSpace| in |jointSpaces|:
+        1. If the tracking system can't provide [=6DoF=] poses for |jointSpace| at the |frame|'s [=XRFrame/time=], set |pose| to `null` and abort these steps..
```

Comment:
Emulating I think is okay, so perhaps say something along the lines of:
"can't provide or emulate poses..."